components/brave_wallet/browser/zcash/zcash_tx_manager.cc

```diff
@@ -71,11 +67,21 @@ void ZCashTxManager::AddUnapprovedTransaction(
         std::move(callback).Run(false, "", "");
         return;
       }
-      zcash_wallet_service_->CreateShieldTransaction(
-          chain_id, from->Clone(), zec_tx_data->to, zec_tx_data->amount, memo,
-          base::BindOnce(&ZCashTxManager::ContinueAddUnapprovedTransaction,
-                         weak_factory_.GetWeakPtr(), chain_id, from.Clone(),
-                         origin, std::move(callback)));
+      if (zec_tx_data->use_shielded_pool) {
+        // Create shielded transaction
+        zcash_wallet_service_->CreateShieldedTransaction(
+            chain_id, from->Clone(), zec_tx_data->to, zec_tx_data->amount, memo,
+            base::BindOnce(&ZCashTxManager::ContinueAddUnapprovedTransaction,
+                           weak_factory_.GetWeakPtr(), chain_id, from.Clone(),
+                           origin, std::move(callback)));
+      } else {
+        // Create transparent to shielded transaction
+        zcash_wallet_service_->CreateShieldTransaction(
+            chain_id, from->Clone(), zec_tx_data->to, zec_tx_data->amount, memo,
```

Comment:
Can it have some better naming?

`CreateShieldedTransaction`
`CreateShieldTransaction`